### PR TITLE
feat(suite-native): only download token definitions for relevant networks

### PR DESCRIPTION
**Pull request by vytick**
- Only definitions for enabled networks and networks of imported accounts are fetched.
 
## Related Issue

Part of #14399

### CODE REVIEW

File: suite-native/discovery/src/discoveryConfigSlice.ts [hunk 1]
```diff
@@ -193,6 +197,15 @@ export const selectDeviceEnabledDiscoveryNetworkSymbols = memoizeWithArgs(
     { size: 2 },
 );
 
+export const selectTokenDefinitionsEnabledNetworks = memoize(
```
Comment (yanascz) [inline suite-native/discovery/src/discoveryConfigSlice.ts:200]: Looking at the selector name, it's not obvious what is being selected... Token definitions or enabled networks? 🤔

Comment (vytick) [reply suite-native/discovery/src/discoveryConfigSlice.ts:200]: I just kept the name used elsewhere

Comment (yanascz) [reply suite-native/discovery/src/discoveryConfigSlice.ts:200]: It's confusing but if it's used elsewhere... 😏

**Review by yanascz — APPROVED** [review]
Seems to work fine! 💪

**Pull request merged**